### ISSUE TITLE
Alias zones sync fails with KeyError

### ISSUE THREAD

**Issue by rissson** [octodns/octodns#633]
Hi everyone!

First, thanks for this great project!

When syncing aliased zones, let's say a zone B. that is aliased to a zone A., if you only sync zone B., without also syncing zone A., it fails with:

```
$ octodns-sync --config-file=./config/config.yaml B.
Traceback (most recent call last):
  File "/home/risson/.cache/pypoetry/virtualenvs/lama-corp-dns-iy2hlWpG-py3.8/bin/octodns-sync", line 8, in <module>
    sys.exit(main())
  File "/home/risson/.cache/pypoetry/virtualenvs/lama-corp-dns-iy2hlWpG-py3.8/lib/python3.8/site-packages/octodns/cmds/sync.py", line 38, in main
    manager.sync(eligible_zones=args.zone, eligible_sources=args.source,
  File "/home/risson/.cache/pypoetry/virtualenvs/lama-corp-dns-iy2hlWpG-py3.8/lib/python3.8/site-packages/octodns/manager.py", line 383, in sync
    desired=desired[zone_source],
KeyError: 'B.'
```

However, `octodns-sync --config-file=./config/config.yaml A. B.` succeeds (the order in which the zones are provided doesn't matter).

A basic example of it failing is already in this repo: `YAML_TMP_DIR=/tmp octodns-sync --config-file=tests/config/simple-alias-zone.yaml alias.tests.`

I'm currently in the process of adding a unit test for this case, please let me know if this behavior is intended. If it isn't, maybe it should be documented that you have to sync the _origin_ zone as well as the aliased one.

A few information:
Python 3.8.6
octoDNS 0.9.11

cc. @ross who added support for this feature

**Comment by ross** [comment]
Nice catch.

> I'm currently in the process of adding a unit test for this case, please let me know if this behavior is intended.

I'd says the behavior is expected, but not having a better error message is an oversight. 

I think the fix would be to throw an error message when an aliased zone is being synced without it companion, e.g. something like:

"A. cannot be synced without B. since it's aliased."

I'm guessing a `try`/`except` around the `KeyError` would be the place to do it.

> documented that you have to sync the _origin_ zone as well as the aliased one.

Yeah. Documenting it would be a good idea as well.

**Comment by rissson** [comment]
Should I add this to #634 then?

**Comment by ross** [comment]
> Should I add this to #634 then?

Yeah. Would be happy to take a look at that.

**Comment by rissson** [comment]
> > Should I add this to #634 then?
> 
> Yeah. Would be happy to take a look at that.

I updated it with the discussed changes.

**Comment by viranch** [comment]
fixed by #634
